nuxeo-runtime/nuxeo-runtime-test/src/test/java/org/nuxeo/runtime/services/config/TestConfigurationService.java

```diff
@@ -113,4 +113,50 @@ public void testCompatWarn() throws Exception {
         Framework.getProperties().remove("nuxeo.test.dummyStringProperty");
     }
 
+    /**
+     * @since 10.3
+     */
+    @Test
+    @Deploy("org.nuxeo.runtime.test.tests:configuration-test-contrib.xml")
+    public void testMerge() throws Exception {
+        assertEquals("dummyValue", cs.getProperty("nuxeo.test.dummyStringProperty"));
+        assertEquals("anotherDummyValue", cs.getProperty("nuxeo.test.anotherDummyStringProperty"));
+        // Deploy another contrib merging existing and overriding properties
+        hotDeployer.deploy("org.nuxeo.runtime.test.tests:configuration-merge-contrib.xml");
+        // Assert new property was merged
+        assertEquals("dummyValue,mergedValue,anotherMergedValue", cs.getProperty("nuxeo.test.dummyStringProperty"));
+        // Assert new property was not merged
+        assertEquals("anotherNotMergedValue", cs.getProperty("nuxeo.test.anotherDummyStringProperty"));
+    }
+
+    /**
+     * @since 10.3
+     */
+    @Test
+    @Deploy("org.nuxeo.runtime.test.tests:configuration-namespace-contrib.xml")
+    public void testByNamespace() throws Exception {
+        assertEquals(3, cs.getProperties("nuxeo.test").size());
+        assertEquals(2, cs.getProperties("nuxeo.anothertest").size());
+        assertEquals(6, cs.getProperties("nuxeo").size());
+        assertEquals(1, cs.getProperties("nuxeo.yetanothertest").size());
+        assertEquals("foo,bar", cs.getProperties("nuxeo.yetanothertest").get("pouet"));
+    }
+
+    /**
+     * @since 10.3
+     */
+    @Test
+    @Deploy("org.nuxeo.runtime.test.tests:configuration-namespace-contrib.xml")
+    public void testToJson() throws Exception {
+        String expected = "{\n" + //
```

Comment:
You can use `org.skyscreamer.jsonassert.JSONAssert` to avoid relying on sorted keys (if the sort below was only for the test).

Reply:
:+1: